mmdet3d/ops/pointnet_modules/point_sa_module.py

```diff
@@ -36,16 +40,30 @@ def __init__(self,
                  norm_cfg: dict = dict(type='BN2d'),
                  use_xyz: bool = True,
                  pool_mod='max',
+                 fps_mod: List[str] = ['D-FPS'],
+                 fps_sample_range_list: List[int] = [-1],
                  normalize_xyz: bool = False):
         super().__init__()
 
         assert len(radii) == len(sample_nums) == len(mlp_channels)
         assert pool_mod in ['max', 'avg']
+        assert isinstance(fps_mod, list) or isinstance(fps_mod, tuple)
+        assert isinstance(fps_sample_range_list, list) or isinstance(
+            fps_sample_range_list, tuple)
+        assert len(fps_mod) == len(fps_sample_range_list)
```

Comment:
assert fps_mod in ['F-FPS', 'D-FPS', 'FS']

Reply:
The assertsion is implemented in the `forward` function

Reply:
It is better to do the assertion at the module initialization thus users could get the assertion earlier and do not need to wait for the model to run inference.